runtime-modules/membership/src/lib.rs

```diff
@@ -1,9 +1,647 @@
+/// Stored information about a registered user
+pub struct Membership<BlockNumber, Moment, PaidTermId, SubscriptionId, AccountId> {
+    /// The unique handle chosen by member
+    pub handle: Vec<u8>,
+
+    /// A Url to member's Avatar image
+    pub avatar_uri: Vec<u8>,
+
+    /// Short text chosen by member to share information about themselves
+    pub about: Vec<u8>,
+
+    /// Block number when member was registered
+    pub registered_at_block: BlockNumber,
+
+    /// Timestamp when member was registered
+    pub registered_at_time: Moment,
+
+    /// How the member was registered
+    pub entry: EntryMethod<PaidTermId, AccountId>,
+
+    /// Whether the member is suspended or not.
+    pub suspended: bool,
+
+    /// The type of subscription the member has purchased if any.
+    pub subscription: Option<SubscriptionId>,
+
+    /// Member's root account id. Only the root account is permitted to set a new root account
+    /// and update the controller account. Other modules may only allow certain actions if
+    /// signed with root account. It is intended to be an account that can remain offline and
+    /// potentially hold a member's funds, and be a source for staking roles.
+    pub root_account: AccountId,
+
+    /// Member's controller account id. This account is intended to be used by
+    /// a member to act under their identity in other modules. It will usually be used more
+    /// online and will have less funds in its balance.
+    pub controller_account: AccountId,
+}
+
+// Contains valid or default user details
+struct ValidatedUserInfo {
+    handle: Vec<u8>,
+    avatar_uri: Vec<u8>,
+    about: Vec<u8>,
+}
+
+#[derive(Encode, Decode, Debug, PartialEq)]
+pub enum EntryMethod<PaidTermId, AccountId> {
+    Paid(PaidTermId),
+    Screening(AccountId),
+    Genesis,
+}
+
+/// Must be default constructible because it indirectly is a value in a storage map.
+/// ***SHOULD NEVER ACTUALLY GET CALLED, IS REQUIRED TO DUE BAD STORAGE MODEL IN SUBSTRATE***
+impl<PaidTermId, AccountId> Default for EntryMethod<PaidTermId, AccountId> {
+    fn default() -> Self {
+        Self::Genesis
+    }
+}
+
+#[derive(Encode, Decode, Eq, PartialEq, Default)]
+pub struct PaidMembershipTerms<Balance> {
+    /// Quantity of native tokens which must be provably burned
+    pub fee: Balance,
+    /// String of capped length describing human readable conditions which are being agreed upon
+    pub text: Vec<u8>,
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Membership {
+        /// MemberId to assign to next member that is added to the registry, and is also the
+        /// total number of members created. MemberIds start at Zero.
+        pub NextMemberId get(members_created) : T::MemberId;
+
+        /// Mapping of member's id to their membership profile
+        pub MembershipById get(membership) : map T::MemberId => MembershipOf<T>;
+
+        /// Mapping of a root account id to vector of member ids it controls.
```

Comment:
These two fields were supposed to be blocked

https://github.com/Joystream/joystream/issues/93#issue-522940475

It's actually a substantial issue because there is non-trivial business logic here and in other modules that depend on the implied (controller & role) account uniqueness, hence removing them will help us track down and update all of those.

Reply:
I incapsulate them to be crate-only: the 'allowed check' is_member() requires them as main components. So, they are blocked. Also, I got rid of their getters, so no one can use this collections from external modules.